### DISCUSSION TITLE
TLS/TCPS Support

### DISCUSSION THREAD

**Discussion by kemra102** [sijms/go-ora#80] [Q&A] [answered]
Is it possible to connect via `tcps` (TLS) instead of plain `tcp`?

I have tried a few things (I'm not a seasoned Go coder) but I couldn't get it working.

**Comment by sijms** [comment]
I think it need more work and until now it is not supported 
but I will do my best to add this feature

**Comment by sijms** [comment]
can you make testing I do a bulk of changes in wallet reading and session SSL support
you database URL should be as follow
```
oracle://user:pass@server:port/service?TRACE FILE=trace.log&wallet=<path_to_wallet_dir>&SSL=true"
```

if you got any error can you share with me

**Reply by kemra102** [reply]
Not 100% sure it's working. I'm getting an error:

```sh
x509: certificate signed by unknown authority
```

The Oracle instance in question is an AWS RDS instance so I guess the cert is signed by AWS and that's not a part of our trust store. Is it possible to skip validation of the cert?

**Comment by sijms** [comment]
Uncomment line no. 221 at session.go
`//InsecureSkipVerify: true, `

and try again

**Comment by sijms** [comment, marked as answer]
I think we are near
I make a commit please check
use the following URL options

`&SSL=TRUE&SSL Verify=TRUE`

the second option to avoid error

`x509: cannot validate certificate for 192.168.1.15 because it doesn't contain any IP SANs`

**Reply by kemra102** [reply]
I'll check this out tomorrow.

**Reply by kemra102** [reply]
So now I get this error:

```sh
tcps need a valid wallet contains server and client certificates
```

Given I'm not looking to validate my certs this seems like an odd error message (not that I really understand what a wallet is).

**Reply by sijms** [reply]
did you pass oracle wallet 
```
wallet=wallet_dir // wallet should contain server and client certificates
SSL=true          // true or enabled
SSL Verify=false  // to bypass certificate verification
```
actually this error message I generate when verify database url
but when I make some testing I find that tcp ssl is working without need for server or  certificate

**Reply by sijms** [reply]
I make commit and remove this restriction

**Reply by kemra102** [reply]
I did try with the wallet but I got an error that quite rightly pointed out that the wallet didn't exist.

I've shut my machine down no, so I'll check again tomorrow with your latest commit.
